### PR TITLE
[cleaner] Add option to skip cleaning files

### PR DESCRIPTION
**Pull request by pmoravec**
A new option --skip-clean-files allows cleaner to skip cleaning files where the user is certain no sensitive information is present.

The option supports globs / wildcards.

Relevant: #3469
Closes: #3520

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [X] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

**Comment by pmoravec** [comment]
I need to add an avocado test before merging.

File: sos/cleaner/__init__.py [hunk 1]
```diff
@@ -262,6 +265,10 @@ def add_parser_options(cls, parser):
                                default=[], dest='disable_parsers',
                                help=('Disable specific parsers, so that those '
                                      'elements are not obfuscated'))
+        clean_grp.add_argument('--skip-clean-files', action='extend',
+                               default=[], dest='skip_clean_files',
+                               help=('List of files to skip/ignore during '
+                                     'cleaning. Asterisks are supported.'))
```
Comment (TurboTurtle) [inline sos/cleaner/__init__.py:271]: s/Asterisks/Globs in my opinion, as it is more precise.

**Review by TurboTurtle — COMMENTED** [review]
Code LGTM, minor notes below.

**Comment by pmoravec** [comment]
I was about to write a test for this when I discovered a problem: we can't easily tell cleaner not to obfuscate symbolic links.

Check:

    sos report --clean -o filesys --keywords=sysfs --batch --build --skip-cleaning-files etc/mtab

Then, https://github.com/sosreport/sos/blob/main/sos/cleaner/__init__.py#L758 causes we skip the decision "shall we obfuscate the file or not?" for a symlink.

We can't easily add "`else:`" branch as the target file of the symlink can be already obfuscated.

I was thinking about this solution: anywere before https://github.com/sosreport/sos/blob/main/sos/cleaner/__init__.py#L676 , traverse the `archive.get_file_list()` for symbolic links. If either symlink matches a parser's `skip_patterns`, add the symlink's `[os.path.realpath](http://docs.python.org/library/os.path.html#os.path.realpath)` target to the parser's `skip_patterns`.

It is bit ugly and lengthy, but I dont see an easier way..

Anyway, that is an independent issue I can deal afterwards.

**Comment by pmoravec** [comment]
Avocado test added, symlinks issue raised as https://github.com/sosreport/sos/issues/3563 .

**Review by TurboTurtle — APPROVED** [review]

**Pull request merged**